frontend/src/hooks/useApi.ts

```diff
@@ -68,6 +72,18 @@ export const useApi = (): PimsAPI => {
     return data;
   };
 
+  axios.isPinAvailable = async (
```

Comment:
I'm wondering if this is the best approach.  This requires making a separate request to determine if the PIN is available.  But you essentially get this information when you submit the property, as the error returned I believe states this.

Reply:
This is true, I guess the tradeoff would be the instant feedback. This is also the same way we check for the PID duplicate. Open to changing this approach though

Reply:
Yeah, I think the bigger question here is how we want to handle problems like this in the future. For every instance of a form validation error that can only be determined by the API, we can
1) Create a new api request to handle that validation
OR
2) We can update the api to return some json that the frontend can use to set errors in formik.

There are some potential issues with #2 that we should consider. In general manually setting errors in formik isn't great, as formik will overwrite those errors the next time validation runs. We would probably have to write some more custom formik code to handle this scenario.

Also, even if we tell the user they have a duplicate on the correct field we aren't able to tell them what values are allowed which may result in the user having to try to submit random values until the error goes away.

Reply:
Personally I'd lean towards #1 if we think there will only be a few instances of this problem across the application. If this is a common issue that will occur on every form, lets consider #2.

Reply:
Generally speaking they shouldn't be entering random numbers.  These numbers are supposed to be generated by another tool, which gives them unique values.  In PIMS they are simply transcribing them.